core/commands/ls.go

```diff
@@ -78,12 +78,13 @@ it contains, with the following format:
 				Links: make([]LsLink, len(dagnode.Links)),
 			}
 			for j, link := range dagnode.Links {
-				link.Node, err = link.GetNode(req.Context(), node.DAG)
+				var linkNode *merkledag.Node
+				linkNode, err = link.GetNode(req.Context(), node.DAG)
```

Comment:
reusing err values across scopes causes a lot of errors, has bitten us in the past already